### PR TITLE
Managed identity to get metrics from Azure

### PR DESCRIPTION
**Pull request by NissesSenap**
Will be used in a kubernetes helm chart

### CODE REVIEW

**Comment by NissesSenap** [comment]
Work in progress, I will verify that I have enough access that I need to perform all my queries.

**Comment by NissesSenap** [comment]
Partly solves: https://github.com/XenitAB/terraform-modules/issues/286

**Comment by NissesSenap** [comment]
I can see metrics in prometheus, this is how they look like.
We could change this to a podmonitor so I will do that tomorrow and fix the changelog.md

```.promql
cluster_autoscaler_cluster_safe_to_autoscale{aggregation="average", container="azure-metrics-exporter", endpoint="http", instance="10.244.2.14:8080", job="azure-metrics-azure-metrics-exporter", namespace="az-metrics", pod="azure-metrics-azure-metrics-exporter-594754789b-xbv4x", resourceGroup="rg-1", resourceID="/subscriptions/000000000000000-11111-222222/resourceGroups/rg-1/providers/Microsoft.ContainerService/managedClusters/aks1", resourceName="aks1", service="azure-metrics-azure-metrics-exporter", subscriptionID="000000000000000-11111-222222", timespan="PT1M", unit="Count"}
```

The question is if we should try to remove some of these labels.

File: modules/kubernetes/aks-core/variables.tf [hunk 1]
```diff
@@ -397,3 +397,22 @@ variable "kube_state_metrics_namespaces_extras" {
...
+  type = object({
+    client_id   = string
+    resource_id = string
+  })
+}
+
+variable "subscription_id" {
+  description = "Current subscription id"
+  type        = string
+}
```
Comment (phillebaba) [inline modules/kubernetes/aks-core/variables.tf:418]: Could we not get this from a datasource?

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,10 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 
 ## Unreleased
 
+### Added
+
+- [#463](https://github.com/XenitAB/terraform-modules/pull/463) Add az-metrics to monitor azure specific metrics
```
Comment (phillebaba) [inline CHANGELOG.md:11]: Please add punctuation

File: modules/azure/aks/outputs.tf [hunk 3]
```diff
@@ -8,3 +8,11 @@ output "kube_config" {
     cluster_ca_certificate = base64decode(azurerm_kubernetes_cluster.this.kube_admin_config[0].cluster_ca_certificate)
   }
 }
+
+output "azure_metrics_identity" {
```
Comment (phillebaba) [inline modules/azure/aks/outputs.tf:12]: Are we calling this az_metrics or azure_metrics? I have a preference of azure_metrics but its not too important, just that we use the same name everywhere.

Comment (NissesSenap) [reply modules/azure/aks/outputs.tf]: I will change to azure instead.

**Review by phillebaba — APPROVED** [review]
LGTM :rose: Really good job!

**Comment by simongottschlag** [comment]
Woop woop! Epic job @NissesSenap 🙌🚀🎉🎉🎉🎉

**Pull request merged**